packages/frontend/src/hooks/dashboard/useDashboardChart.ts

```diff
@@ -1,27 +1,193 @@
+    context?: string;
+}) => {
+    return lightdashApi<ApiChartAndResults>({
+        url: `/saved/${chartUuid}/chart-and-results${
+            context ? `?context=${context}` : ''
+        }`,
+        method: 'POST',
+        body: JSON.stringify({
+            dashboardUuid,
+            dashboardFilters,
+            dashboardSorts,
+            granularity,
+            ...(invalidateCache && { invalidateCache: true }),
+            autoRefresh,
+        }),
+    });
+};
+
+const useDashboardChart = (tileUuid: string, chartUuid: string | null) => {
+    const queryPaginationEnabled = useFeatureFlagEnabled(
+        FeatureFlags.QueryPagination,
+    );
+    console.log('queryPaginationEnabled 1', queryPaginationEnabled);
     const dashboardUuid = useDashboardContext((c) => c.dashboard?.uuid);
     const invalidateCache = useDashboardContext((c) => c.invalidateCache);
     const dashboardFilters = useDashboardFiltersForTile(tileUuid);
     const chartSort = useDashboardContext((c) => c.chartSort);
-    const tileSort = chartSort[tileUuid] || [];
+    const dashboardSorts = useMemo(
+        () => chartSort[tileUuid] || [],
+        [chartSort, tileUuid],
+    );
     const granularity = useDashboardContext((c) => c.dateZoomGranularity);
-    const isAutoRefresh = useDashboardContext((c) => c.isAutoRefresh);
-    const context = useSearchParams('context') || undefined;
-    return useChartAndResults(
-        savedChartUuid,
-        dashboardUuid ?? null,
-        dashboardFilters,
-        tileSort,
+    const autoRefresh = useDashboardContext((c) => c.isAutoRefresh);
+    const context =
+        useSearchParams<QueryExecutionContext>('context') || undefined;
+    const setChartsWithDateZoomApplied = useDashboardContext(
+        (c) => c.setChartsWithDateZoomApplied,
+    );
+    const queryClient = useQueryClient();
+
+    const sortKey =
+        dashboardSorts
+            ?.map((ds) => `${ds.fieldId}.${ds.descending}`)
+            ?.join(',') || '';
+    const queryKey = useMemo(
+        () => [
+            'savedChartResults',
+            queryPaginationEnabled,
+            chartUuid,
+            dashboardUuid,
+            dashboardFilters,
+            invalidateCache,
+            sortKey,
+            autoRefresh,
+        ],
+        [
+            queryPaginationEnabled,
+            chartUuid,
+            dashboardUuid,
+            dashboardFilters,
+            invalidateCache,
+            sortKey,
+            autoRefresh,
+        ],
+    );
+    const apiChartAndResults =
+        queryClient.getQueryData<ApiChartAndResults>(queryKey);
+
+    const timezoneFixFilters =
+        dashboardFilters && convertDateDashboardFilters(dashboardFilters);
+    const hasADateDimension =
+        !!apiChartAndResults?.metricQuery?.metadata?.hasADateDimension;
+
+    const fetchChartAndResults = useCallback<
+        () => Promise<ApiChartAndResults>
+    >(async () => {
+        if (queryPaginationEnabled) {
```

Comment:
This section is the new code. The rest was moved from packages/frontend/src/hooks/useQueryResults.ts

Reply:
Instead of using the chart-and-results endpoint we now need 3 requests: chart, explore, results.